src/core/src/main/java/org/apache/jmeter/gui/action/SelectTemplatesDialog.java

```diff
@@ -260,7 +261,9 @@ public void windowClosing(WindowEvent evt){
 
         this.pack();
         this.setMinimumSize(new Dimension(MINIMAL_BOX_WIDTH, MINIMAL_BOX_HEIGHT));
-        ComponentUtil.centerComponentInWindow(this, 50); // center position and 50% of screen size
+        SwingUtilities.invokeLater(() ->
+            ComponentUtil.centerComponentInWindow(this, 50) // center position and 50% of screen size
+        );
         populateTemplatePage();
```

Comment:
populateTemplatePage seems to be a UI method as well.
WDYT of moving it inside `invokeLater` above?

Reply:
Yes, I thought of that and I thought about moving it up in the code, as it is setText and getText, only. That should be safe, shouldn't it?

Reply:
I've no idea. I would move it just in case. However, it might look all the others need to be on the UI thread. Should we create the class on the UI thread in the first place?

Reply:
I adapted JMeterTest, to create all classes in the AWT thread. That is probably a bit saner.

Reply:
Indeed, creating all the instances in AWT thread looks better in that test.